File: app/src/main/java/org/wikipedia/dataclient/okhttp/util/HttpUrlUtil.java
package org.wikipedia.dataclient.okhttp.util;

import android.support.annotation.NonNull;

import java.util.Arrays;
import java.util.Collections;
import java.util.List;

import okhttp3.HttpUrl;

public final class HttpUrlUtil {
    private static final List<String> RESTBASE_SEGMENT_IDENTIFIERS = Arrays.asList("rest_v1", "v1");

    public static boolean isRestBase(@NonNull HttpUrl url) {
        return !Collections.disjoint(url.encodedPathSegments(), RESTBASE_SEGMENT_IDENTIFIERS);
    }

    public static boolean isMobileView(@NonNull HttpUrl url) {
        return "mobileview".equals(url.queryParameter("action"));
    }

    private HttpUrlUtil() { }
}

File: app/src/main/java/org/wikipedia/dataclient/okhttp/util/HttpUrlUtil.kt
package org.wikipedia.dataclient.okhttp.util

import okhttp3.HttpUrl
import java.util.Collections

object HttpUrlUtil {
    private val RESTBASE_SEGMENT_IDENTIFIERS = listOf("rest_v1", "v1")

    @JvmStatic
    fun isRestBase(url: HttpUrl): Boolean {
        return !Collections.disjoint(url.encodedPathSegments, RESTBASE_SEGMENT_IDENTIFIERS)
    }

    @JvmStatic
    fun isMobileView(url: HttpUrl): Boolean {
        return "mobileview" == url.queryParameter("action")
    }
}
